db/updateItem.go

```diff
@@ -39,8 +41,46 @@ func (d *dynamoDB) UpdateItem(listID string, itemID string, newName string) (*da
 		if e.Code() == dynamodb.ErrCodeConditionalCheckFailedException {
 			return nil, ErrorNotFound
 		}
+		if e.Code() == "ValidationException" { // https://github.com/aws/aws-sdk-go/issues/3140
+			return nil, ErrorBadRequest
+		}
+		return nil, err
 	default:
 		return nil, err
 	}
-	return item, nil
+
+	item := new(data.Item)
+	err = dynamodbattribute.UnmarshalMap(output.Attributes, &item)
+	return item, err
+}
+
+func getUpdateFields(newName string, isCompleted *bool) (map[string]*dynamodb.AttributeValue, *string, map[string]*string) {
+	fields := map[string]*dynamodb.AttributeValue{}
+	names := map[string]*string{}
+	var updateExpression *string
+
+	if isCompleted != nil {
+		fields[":c"] = &dynamodb.AttributeValue{BOOL: isCompleted}
+		updateExpression = appendUpdateExpression(updateExpression, "IsCompleted = :c")
+	}
+
+	if newName != "" {
+		fields[":n"] = &dynamodb.AttributeValue{S: aws.String(newName)}
+		names["#n"] = aws.String("Name")
+		updateExpression = appendUpdateExpression(updateExpression, "#n = :n")
```

Comment:
Since `NAME` is a reserved keyword in dynamodb we have to do an extra step to make changes to it:
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ReservedWords.html